### PR TITLE
Fix "Unknown Channel" bug on CustomReactors

### LINKED ISSUE #6289: Custom Reactors can't communicate due to unknown Channels
**Tendermint version**: d988cd61018a7ae3884762d4e5188df1c67833c1

**What happened**:
When setting a custom reactor on `NewNode` using `CustomReactors` options with a custom name (not `MEMPOOL` ... `STATESYNC`), they are not able to communicate due to a missing channel error:

```
D[2021-03-29|11:51:09.606] Unknown channel for peer                     channel=128 channels=40202122233038606100
```

**What you expected to happen**:
I'd expect the reactor to be able to send and receive messages between peers.

**Have you tried the latest version**: yes

**How to reproduce it** (as minimally and precisely as possible):
Here's my reactor code:
```go
const (
	Channel = byte(0x80)
)

type Reactor struct {
	p2p.BaseReactor
}

func NewReactor() *Reactor {
	r := &Reactor{}
	r.BaseReactor = *p2p.NewBaseReactor("MyReactor", r)
	return r
}

func (r *Reactor) GetChannels() []*conn.ChannelDescriptor {
	return []*conn.ChannelDescriptor{
		{ID: Channel, Priority: 2},
	}
}

func (r *Reactor) InitPeer(peer p2p.Peer) p2p.Peer {
	peer.SetLogger(
		log.NewTMLogger(os.Stdout),
	)
	return peer
}

func (r *Reactor) AddPeer(peer p2p.Peer) {
	peer.Send(Channel, []byte("hello"))
}

func (r *Reactor) Receive(chID byte, peer p2p.Peer, msgBytes []byte) {
	fmt.Printf("msgBytes = %+v\n", msgBytes)
}
```

### PR DESCRIPTION
**Pull request by gchaincl**
New Peers are created passing a list of static channels.
When adding a new CustomReactor to NewNode, the new channel(s) used by
that reactor were not passed to the new peers, creating an error when
peers tried to communicate.

Closes #6289

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

**Comment by alexanderbez** [comment]
@marbar3778 do you think we need a changelog for this?

**Comment by marbar3778** [comment]
> @marbar3778 do you think we need a changelog for this?

yes, it does alter some previous assumed behavior even though no one may have been using it.

**Comment by gchaincl** [comment]
> > @marbar3778 do you think we need a changelog for this?
> 
> yes, it does alter some previous assumed behavior even though no one may have been using it.

I just added the entry in CHANGELOG_PENDING.md

File: CHANGELOG_PENDING.md [hunk 1]
```diff
@@ -28,3 +28,5 @@ Friendly reminder, we have a [bug bounty program](https://hackerone.com/tendermi
 ### BUG FIXES
 
 - [rpc/jsonrpc/server] \#6191 Correctly unmarshal `RPCRequest` when data is `null` (@melekes)
+
```
Comment (alexanderbez) [inline CHANGELOG_PENDING.md:31]:
```suggestion
```

**Comment by kostko** [comment]
I believe this fix is incorrect. It seems that previously it used the peer's list of channels but now it uses the local node's list of channels in its place. In the default configuration these would be the same, but for example if you have a PEX-only seed node on one end (or any two peers where the channels don't match exactly), this will break.

Since `peer.channels` is supposed to be the channels that the peer announces to support during the connection handshake (exchanged via the `NodeInfo` structure) this change will make `hasChannel` incorrectly return `true` which will cause peers to send messages over channels that the other peer does not necessarily support. And the other peer will tear down the connection since sending on a channel that the peer did not announce is a protocol violation.

**Pull request merged**